northstar/src/runtime/state.rs

```diff
@@ -400,10 +400,12 @@ impl State {
 
     /// Install an NPK
     pub async fn install(&mut self, id: &str, npk: &Path) -> Result<(), Error> {
-        let repository = self
-            .repositories
-            .get(id)
-            .ok_or_else(|| Error::RepositoryNotFound(id.to_owned()))?;
+        let repository = self.repositories.get(id).ok_or_else(|| {
+            Error::RepositoryIdUnknown(
+                id.to_owned(),
+                self.repositories.keys().map(|id| id.into()).collect(),
```

Comment:
is it ok then computing this every time the error is generated?

Reply:
didn't see this was answered somewhere :) my bad

Reply:
I think this is a much simple solution like it is. And I'd say simplicity outweighs the cost (which is negligible here)